lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -2391,16 +2404,27 @@ export class UserAgentApplication {
   }
 
   /**
-   * Remove sid and login_hint if passed as extraQueryParameters
+   * Removes unnecessary or duplicate query parameters from extraQueryParameters
+   * @param request
    * @param eQParams
    */
-  private removeSSOParamsFromEQParams(eQParams: QPDict): QPDict {
-
-    if (eQParams) {
+  private sanitizeEQParams(request: AuthenticationParameters) : QPDict {
+    let eQParams : QPDict = request.extraQueryParameters;
+    if (!eQParams) {
+      return null;
+    }
+    if (request.claimsRequest) {
+      this.logger.warning("Removed duplicate claims from extraQueryParameters. Please use either the claimsRequest field OR pass as extraQueryParameter - not both.");
+      delete eQParams[Constants.claims];
+    }
+    if (request.sid) {
```

Comment:
Please remove the if check. The idea is to remove any sid, login_hint (and may be claims too) from extraQueryParameters as we do not support the pattern anymore. 

Also this check will be a problem for anyone passing sid by accident in extraQueryParameters and contrasts with our library populating the hints in addHintParameters.